db/compaction/compaction_picker_level.cc

```diff
@@ -281,7 +281,13 @@ void LevelCompactionBuilder::SetupInitialFiles() {
   }
 
   // Periodic Compaction
-  PickFileToCompact(vstorage_->FilesMarkedForPeriodicCompaction(), false);
+  bool periodic_compact_to_next_level =
+      ioptions_.level_compaction_dynamic_level_bytes &&
+      (mutable_cf_options_.ttl == 0 /* ttl was disabled */ ||
+       mutable_cf_options_.ttl >
+           mutable_cf_options_.periodic_compaction_seconds);
+  PickFileToCompact(vstorage_->FilesMarkedForPeriodicCompaction(),
+                    periodic_compact_to_next_level);
```

Comment:
I was hoping that the trigger would be "data age exceeds periodic_compaction_seconds", like TTL compaction. The data age (or `oldest_ancester_time`) is a bit different from the file age. If we use file age, then, for example, to periodically compact from L1->L6, `5*periodic_compaction_seconds` would need to elapse. Whereas, using data age, only `periodic_compaction_seconds` would need to elapse, or less if the L1 data is not fresh to begin with.

Implementation-wise, I suspect reusing the `ComputeExpiredTtlFiles()` would be a simple way to achieve this.

Also, we will need to update the release notes to mention this behavior change. See `unreleased_history/README.txt` or adapt from examples in that directory.

Reply:
> If we use file age, then, for example, to periodically compact from L1->L6, 5*periodic_compaction_seconds would need to elapse

Discussed with @cbi42. This is fine with us.